### PR TITLE
Polishing R Script ahead of LHD Meeting

### PR DESCRIPTION
**Pull request by apulsipher**
Implements major code polishing and safety features ahead of meeting with LHD partners.

### CODE REVIEW

File: tricounty-report.R [hunk 1]
```diff
@@ -1,216 +1,329 @@
...
+  
+  # - Sort alphabetically so missing diseases are correctly placed
+  monthly_avgs <- monthly_avgs[order(monthly_avgs$disease),]
+}
+
+# - Find the previous month of the report year
+report_month <- max(month_counts[month_counts$year == report_year, ]$month) - 1
 
 # create_public_report() creates a public report for the given month
 create_public_report <- function(month_num) {
```
Comment (gvegayon) [inline tricounty-report.R:290]: Pass the needed data as an argument; avoid using globals/scoping.

File: tricounty-report.R [hunk 2]
```diff
@@ -1,216 +1,329 @@
...
-
-# - Create report table for two months prior
-create_public_report(current_report_month - 2)
-
-
-
+# - Create report table for most recent month and for 1 and 2 months prior
+create_public_report(report_month)
+create_public_report(report_month - 1)
+create_public_report(report_month - 2)
```
Comment (gvegayon) [inline tricounty-report.R:329]: See the comment on passing objects as function arguments instead of using scoping.

**Review by gvegayon — CHANGES_REQUESTED** [review]
Looking great, @apulsipher! A general comment, ensure you are documenting the function using `roxygen2`. Regardless of whether we use it as a package or not in the future. Another thing I think we need is to have a dummy dataset with a couple of tests: 

- A few records only, ensuring that you match the averages you are expecting.
- A couple of error tests in which you provide the wrong data.
- A couple of edge cases in which you have too few points (.e.g, less than 6 years of data) and another in which data has missings or something like that.

We can add more coverage later.

File: tricounty-report.R [hunk 3]
```diff
@@ -1,216 +1,342 @@
...
+
+# ------------------------------------------------------------------------------
+# Define Helper Functions ------------------------------------------------------
+
+# Writes report CSV files
+write_report_csv <- function(data, filename, is.public = FALSE) {
+  write.csv(data,
+            file.path(ifelse(is.public, 
+                             public_reports_folder, 
+                             internal_reports_folder),
```
Comment (gvegayon) [inline tricounty-report.R:34]: Again, be careful with using too much scoping.

File: tricounty-report.R [hunk 4]
```diff
@@ -1,216 +1,329 @@
...
+}
 
-# Rearrange columns (for DEBUGGING)
-epitrax_data <- epitrax_data[,c("disease", "month", "year", "counts")]
+# ------------------------------------------------------------------------------
+# Read in EpiTrax data ---------------------------------------------------------
+epitrax_data <- read_epitrax_data()
+epitrax_data_yrs <- sort(unique(epitrax_data$year))
+epitrax_data_diseases <- unique(epitrax_data$disease)
+report_year <- max(epitrax_data_yrs)
```
[the remark this answers is not in the document]
Comment (apulsipher) [reply tricounty-report.R]: I like this idea, but I might wait until I see how the app evolves before implementing it. I'll move it to a separate issue.
